### ISSUE TITLE
`clippy::write_literal` should suggest using `write_str` for `fmt::Write`

### ISSUE THREAD

**Issue by maia-s** [rust-lang/rust-clippy#13449]
### Summary

The `clippy::write_literal` lint suggests changing code like `write!(f, "{}", "{}")` to `write!(f, "{{}}")`, but this can require a lot of changes to the string if it contains many `{` or `}` chars. If the first argument to `write!` implements `fmt::Write`, clippy should suggest changing it to `f.write_str("{}")` instead.

### Reproducer

I tried this code:

```rust
use std::fmt::Write;
fn main() {
    let mut s = String::new();
    write!(s, "{}", "{}").unwrap();
}
```

I expected to see this happen:
```
warning: literal with an empty format string
 --> clippy-write-literal.rs:4:21
  |
4 |     write!(s, "{}", "{}").unwrap();
  |                     ^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_literal
  = note: `#[warn(clippy::write_literal)]` on by default
help: try
  |
4 -     write!(s, "{}", "{}").unwrap();
4 +     s.write_str("{}").unwrap();
  |

warning: 1 warning emitted
```

Instead, this happened:
```
warning: literal with an empty format string
 --> clippy-write-literal.rs:4:21
  |
4 |     write!(s, "{}", "{}").unwrap();
  |                     ^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#write_literal
  = note: `#[warn(clippy::write_literal)]` on by default
help: try
  |
4 -     write!(s, "{}", "{}").unwrap();
4 +     write!(s, "{{}}").unwrap();
  |

warning: 1 warning emitted
```

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: aarch64-apple-darwin
release: 1.81.0
LLVM version: 18.1.7

**Comment by ada4a** [comment]
I wouldn't say this is a bug necessarily, but rather a suboptimal suggestion.

@rustbot label -C-bug +C-enhancement

**Comment by ada4a** [comment]
@rustbot label S-blocked

because of [#clippy > How to suggest importing a path](https://rust-lang.zulipchat.com/#narrow/channel/257328-clippy/topic/How.20to.20suggest.20importing.20a.20path/with/544195729)
